### PR TITLE
[ROCm] Fix for ROCm CSB breakage - 210311

### PR DESCRIPTION
**Pull request by deven-amd**
The following commit breaks the ROCm CSB (errors while running GPU unit tests)

https://github.com/tensorflow/tensorflow/pull/47398

```
ERROR: /root/tensorflow/tensorflow/compiler/xla/service/gpu/tests/BUILD:573:13: undeclared inclusion(s) in rule '//tensorflow/compiler/xla/service/gpu/tests:hlo_to_llvm_ir':
this rule is missing dependency declarations for the following files included by 'tensorflow/compiler/xla/service/gpu/tests/hlo_to_llvm_ir.cc':
  'tensorflow/compiler/xla/service/gpu/nvptx_compiler.h'
Target //tensorflow/compiler/xla/service/gpu/tests:constant.hlo.test failed to build
```

This PR/commit fixes the ROCm build, by making the newly added functionality specific to the TF CUDA build.

Long term, we need to add similar capability to the ROCM side, but for now this workaround is needed to get the ROCm CSB working again.

----------------------------------

/cc @cheshire @chsigg @penpornk

### CODE REVIEW

**Comment by cheshire** [comment]
It's up to you how to do this. I'll approve, but it's a bit annoying to lose this coverage for ROCm, and ideally the target should be separated into the common part and the NVIDIA-specific part.

**Comment by cheshire** [comment]
ah, actually you did this already. Great!

File: tensorflow/compiler/xla/service/gpu/tests/BUILD [hunk 1]
```diff
@@ -594,6 +594,12 @@ glob_lit_tests(
         "no_pip",
     ],
     driver = "@llvm-project//mlir:run_lit.sh",
+    tags_override = {
+        "reduction_vectorization_sm50.hlo": ["no_rocm"],
```
Comment (cheshire) [inline tensorflow/compiler/xla/service/gpu/tests/BUILD:598]: Maybe rename those targets as well to make it explicit that they are NVIDIA-specific and test the output PTX?

[Also is there a way to test vectorization on ROCm in the same way?]

**Review by cheshire — APPROVED** [review]

File: tensorflow/compiler/xla/service/gpu/tests/BUILD (hunk 1 above)
Comment (deven-amd) [reply tensorflow/compiler/xla/service/gpu/tests/BUILD:598]: >[Also is there a way to test vectorization on ROCm in the same way?]

There should be in theory. Let me do some digging and figure it out

**Comment by gbaned** [comment]
@deven-amd Can you please resolve conflicts? Thanks!

**Comment by deven-amd** [comment]
@gbaned The PR ( https://github.com/tensorflow/tensorflow/pull/47398) that broke ROCm build (and thus required this fix) seems to have been reverted ( https://github.com/tensorflow/tensorflow/commit/99b9d93780d68190ba0f7b8c3b97c665fc63077b) . 

So it seems that this PR is no longer required. Do you want me to close this PR, or leave it around because we will need it once PR #47398  is admitted again?

@mihaimaruseac @sanjoy @nouiz j just FYI. 
Any chance, we can make the commits in this PR, part of PR #47398 when it gets re-submitted? That way we can ensure it does not break the ROCm build to begin with.

thanks

**Comment by mihaimaruseac** [comment]
Can we merge this with #47398 (cloned for resubmission)?

**Comment by nouiz** [comment]
I re-reverted my original MR, then applied your commits and then applied the missing commits from my original MR:

https://github.com/tensorflow/tensorflow/pull/47707/

@mihaimaruseac can you confirm that it works on your side, so then when they merge it, it hopefully won't break their internal tests? (Note, I'm not sure if their internal failure is your or something else, I didn't see the internal error).

**Comment by gbaned** [comment]
@deven-amd Can you please resolve conflicts? Thanks!

**Comment by deven-amd** [comment]
Closing out this PR as the commits in this PR were merged in as part of #47707 

thank you @nouiz

**Pull request closed without merging**